services/contract_call_local.proto

```diff
@@ -134,6 +134,16 @@ message ContractCallLocalQuery {
      * max number of bytes that the result might include. The run will fail if it would have returned more than this number of bytes.
      */
     int64 maxResultSize = 5 [deprecated=true];
+
+    /**
+     * List with accounts that should be "warmed up"/loaded in the EVM
```

Comment:
If I understand correctly, it sounds like this feature is used to pre-cache certain data in the EVM so contract calls are faster for people that pay for the privilege. This makes sense to have on a contract create or update.

So I'm confused on how these can be utilized on a contract call or contract call local query. When you're already executing a contract there's no chance to warm it up. If you're saying that the next call will see the warm data that might make sense for contract call but not contract call local since it should not modify state (outside payment).

Even for contract call I think the access list is something that's more associated with the contract entity and not for every execution of the bytecode. It seems like these access list modifications would be better suited to only be on contract create or update.

Reply:
I think the subtlety that is being missed is that the contract calls and slot accesses can be highly variable based on  function call parameters.

The warm up is to bulk query the slot data and account data so that when it is read it is in memory instead of on disk. So when a sub-call, or account read, or slot read happens we don't have to stall for disk access.  Especially querying the code for CALL operations. This is what's killing us the most on the benchmarks pre-virtual-merkle: getting contract code for calls off of the postgres DB.

Another piece of the puzzle is in q2-q3 next year I expect the gas schedules will be changed to make pre-cahced calls always cheaper and in most cases much cheaper than not pre-caching.

The final piece of the puzzle is that this is a path to parallel execution of calls.  Though not as useful in queries since there is no write conflict we can use this data, if we lock in the requirement that access lists limit reads and writes then we can run multiple threads and guarantee there will be no write conflicts with parallel threads.

For the Ethereum perspective see [EIP-2930](https://eips.ethereum.org/EIPS/eip-2930), this is a fairly direct mapping of that spec to hederea.

Reply:
Ah, okay that makes more sense. Thanks for explaining.